packages/salesforcedx-vscode-apex/src/testDiscovery/testDiscoveryService.ts

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import type { ToolingTestClass, TestDiscoveryResult, ToolingTestsPage } from './schemas';
+import type { Connection } from '@salesforce/core';
+import { getVscodeCoreExtension } from '../coreExtensionUtils';
+import { getTelemetryService } from '../telemetry/telemetry';
+
+/**
+ * Service that retrieves Apex test classes and methods from the Tooling REST Test Discovery API.
+ * Docs: https://developer.salesforce.com/docs/atlas.en-us.api_tooling.meta/api_tooling/intro_rest_resources_testing_discovery.htm
+ */
+export class TestDiscoveryService {
```

Comment:
why make it a class?

Are there scenarios where you'd want to create it but not call discover and just pass in the connection?  [alternatively, is discover the only thing you can do and therefore it should just be a fn]



should the ctor private (like you want consumers to only create it via `create` and not to `new` it directly?)